### PR TITLE
Remove exporter module

### PR DESCRIPTION
**Pull request by heyams**

### CODE REVIEW

File: agent/agent-tooling/build.gradle.kts [hunk 1]
```diff
@@ -19,11 +19,9 @@ dependencies {
   implementation(project(":agent:agent-gc-monitor:gc-monitor-core"))
   implementation(project(":agent:agent-profiler:agent-diagnostics-api"))
   implementation(project(":agent:agent-profiler:agent-diagnostics"))
-  implementation(project(":agent:azure-monitor-exporter")) {
-    exclude("org.ow2.asm", "asm")
```
Comment (heyams) [inline agent/agent-tooling/build.gradle.kts:23]: @trask not sure about this exclude

File: buildSrc/src/main/kotlin/ai.errorprone-conventions.gradle.kts [hunk 2]
```diff
@@ -59,9 +59,6 @@ tasks {
         disable("UnnecessaryDefaultInEnumSwitch")
         disable("InconsistentOverloads")
 
-        // consider enabling this after removing azure-monitor-exporter from this repo
-        disable("CanIgnoreReturnValueSuggester")
```
Comment (heyams) [inline buildSrc/src/main/kotlin/ai.errorprone-conventions.gradle.kts:63]: it seems this is still needed.

File: agent/agent-tooling/build.gradle.kts (hunk 1 above)
Comment (trask) [reply agent/agent-tooling/build.gradle.kts:23]: check `./gradlew :agent:agent-tooling:dependencies` to see if it's a transitive dependency

File: buildSrc/src/main/kotlin/ai.errorprone-conventions.gradle.kts (hunk 2 above)
Comment (trask) [reply buildSrc/src/main/kotlin/ai.errorprone-conventions.gradle.kts:63]: that's ok to keep it (and remove the comment)

**Review by trask — APPROVED** [review]

**Pull request merged**